### ISSUE TITLE
[Bug]: Meter chart legend overlaps with chart area when width decreased

### ISSUE THREAD

**Issue by oetlinge** [carbon-design-system/carbon-charts#1616]
### Name

Michael Oetlinger

### Are you an IBM employee?

- [X] Yes

### What happened?

When creating a proportional meter chart, resizing the window causes the legend area of the chart to overlap with the chart itself when the width is sufficiently decreased. Was able to reproduce this issue using the proportional meter chart on the carbon charts angular storybook page (https://charts.carbondesignsystem.com/angular/?path=/story/simple-charts-meter-proportional--proportional-meter-chart) as well as on stackblitz sandbox (https://angular-gi46tp.stackblitz.io)

For comparison, here's what the chart looks like when fully displayed:
![image](https://github.com/carbon-design-system/carbon-charts/assets/59981506/fabd9468-3d60-4862-b0cc-b4dbc19eb473)

Here's the same chart when the window's width is decreased:
![image](https://github.com/carbon-design-system/carbon-charts/assets/59981506/3f78ee5f-ac66-43d5-8d33-c87726040df3)

Note how the text in the chart is also obscured by the overlapping legend area.

### Version

`@carbon/charts@1.11.8`
`@carbon/charts-angular@1.11.8`

### Data & options used

```js
Data:
[
  {
    "group": "emails",
    "value": 202
  },
  {
    "group": "photos",
    "value": 654
  },
  {
    "group": "text messages",
    "value": 723
  },
  {
    "group": "videos",
    "value": 120
  },
  {
    "group": "emails2",
    "value": 202
  },
  {
    "group": "photos2",
    "value": 654
  },
  {
    "group": "text messages2",
    "value": 723
  },
  {
    "group": "videos2",
    "value": 120
  }
]

Options:
{
  "title": "Proportional Meter Chart",
  "height": "130px",
  "meter": {
    "proportional": {
      "total": 2000,
      "unit": "GB"
    }
  },
  "color": {
    "pairing": {
      "option": 2
    }
  }
}
```

### Codesandbox example

https://stackblitz.com/edit/angular-gi46tp

### What priority level would this be in your opinion?

P3

**Comment by theiliad** [comment]
Hi, what would be the expected behavior here? it seems like we're not providing it with enough of a vertical area to place everything

**Comment by oetlinge** [comment]
If there's an explicit limitation set on the height that prevents wrapping the legend items to additional rows, then one possible solution might be to add a vertical scrollbar. Another possibility could be to limit the minimum width of the chart so that it never gets forced to overlap.

**Comment by theiliad** [comment]
Yes, I'm wondering what we can do **automatically** on our end. It's difficult to anticipate this on the library side...

I've also just tried doing a similar thing in Recharts, and I'm seeing similar behavior

### height set at 300
<img width="574" alt="image" src="https://github.com/carbon-design-system/carbon-charts/assets/14989804/44857868-4388-44da-a630-7de0e516c6dd">

### height set at 200
<img width="571" alt="image" src="https://github.com/carbon-design-system/carbon-charts/assets/14989804/ccff491b-f49a-4074-a7bd-b8c304b87f2e">

### height set at 100
<img width="571" alt="image" src="https://github.com/carbon-design-system/carbon-charts/assets/14989804/38378283-3278-4a30-8190-4af8535a4d4a">

**Comment by ab1109** [comment]
Hi I am also facing a similar problem for the proportional meter chart when the window's width is decreased as shown above. Would there be a fix available for this issue?
